driver/tests/test_restart.py

```diff
@@ -0,0 +1,72 @@
+import unittest.mock
+
+import numpy as np
+import xarray as xr
+import yaml
+import zarr
+
+import pace.dsl
+from pace.driver import DriverConfig
+from pace.driver.state import DriverState
+
+from .test_driver import NullCommConfig
+
+
+def test_save_restart():
+    driver_config = DriverConfig(
+        stencil_config=pace.dsl.StencilConfig(),
+        nx_tile=12,
+        nz=79,
+        dt_atmos=225,
+        days=0,
+        hours=0,
+        minutes=0,
+        seconds=225,
+        layout=(1, 1),
+        initialization=unittest.mock.MagicMock(),
+        performance_config=unittest.mock.MagicMock(),
+        comm_config=NullCommConfig((1, 1)),
+        diagnostics_config=unittest.mock.MagicMock(),
+        dycore_config=unittest.mock.MagicMock(nwat=6),
+        physics_config=unittest.mock.MagicMock(),
+        save_restart=True,
+    )
+    assert driver_config.save_restart is True
```

Comment:
I think this is too trivial of a test to include for how complex it is (lots of mocks needed for DriverConfig), here we're basically testing that dataclasses work.

If you want to test the default behavior is to save restarts (or that it's not to save restarts) that would make sense but you'll want to remove it from the keyword arguments passed if so.

Reply:
Changed to testing default value (should be False)